Dockerfile

```diff
@@ -0,0 +1,61 @@
+FROM node:16-alpine as base
+# Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
+RUN apk add --no-cache libc6-compat
+
+ARG PNPM_VERSION=7.1.1 \
+    APP \
+    # Next.js collects completely anonymous telemetry data about general usage.
+    # Learn more here: https://nextjs.org/telemetry
+    NEXT_TELEMETRY_DISABLED=1
+
+ENV APP=${APP} \
+    NEXT_TELEMETRY_DISABLED=${NEXT_TELEMETRY_DISABLED}
+
+WORKDIR /workspace
+
+RUN corepack enable && corepack prepare pnpm@${PNPM_VERSION} --activate
+
+COPY pnpm-lock.yaml .
+RUN pnpm fetch
+
+FROM base as dev
+
+COPY *.yaml .
+COPY *.json .
+COPY packages ./packages
+COPY apps/${APP} ./apps/${APP}
+
+RUN pnpm --filter "${APP}" install --frozen-lockfile
+RUN pnpm --filter "${APP}" build
```

Comment:
@kilemensi  I can update this although we have already performed the heavy lifting I think by only including files of the app we are building no?
Also, if we intend to use this approach we will have to remove this approach here 
```
RUN pnpm --filter "${APP}" install --frozen-lockfile
```
and install all dev dependencies.
I feel like `turbo` would help when we have multiple repos and but in this case, we have already removed/ignored all the other repos.

Reply:
1. apps don't depend on each; they depend on packages. Why would we need to install anything more that what a specific app needs.
2. Include `--offline` on `install` (Otherwise, there is no benefit of [`pnpm fetch`](https://pnpm.io/cli/fetch))
3. `turbo` [benefits](https://turborepo.org/docs#why-turborepo).